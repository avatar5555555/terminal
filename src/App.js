// @flow
import React, { Fragment } from 'react'
import { ToastContainer } from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css'

import Router from 'src/router'
import Bar from 'src/Components/Bar'

const App = () => (
  <Fragment>
    <Bar />
    <Router />
    <ToastContainer />
  </Fragment>
)

export default App
