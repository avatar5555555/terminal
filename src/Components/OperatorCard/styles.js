// @flow

import type { Theme } from 'src/types'

const styles = (theme: Theme) => ({
  image: {
    width: 100,
    height: 100,
    margin: theme.spacing.unit
  },
  container: {
    display: 'flex',
    flex: 1
  },
  root: {
    width: '100%'
  },
  icon: {
    marginLeft: theme.spacing.unit
  },
  button: {
    alignSelf: 'center',
    [theme.breakpoints.down('sm')]: {
      alignSelf: 'flex-end'
    }
  },
  content: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    flex: 1,
    [theme.breakpoints.down('sm')]: {
      flexDirection: 'column',
      alignItems: 'flex-end'
    }
  }
})

export default styles
