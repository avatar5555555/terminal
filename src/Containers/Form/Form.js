// @flow

import React, { Component } from 'react'
import Grid from '@material-ui/core/Grid'
import InputAdornment from '@material-ui/core/InputAdornment'
import MenuItem from '@material-ui/core/MenuItem'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'
import Payment from '@material-ui/icons/Payment'
import CircularProgress from '@material-ui/core/CircularProgress'

import MaskedInput from 'src/Components/MaskedInput'
import type { FormProps } from 'src/types'

class Form extends Component<FormProps> {
  componentDidMount() {
    this.props.fetchList()
  }

  getProps = (id: string) => {
    const { errors, touched, values, handleChange, handleBlur } = this.props
    return {
      id,
      onChange: handleChange,
      onBlur: handleBlur,
      valid: Boolean(!errors[id] || !touched[id]),
      error: errors[id],
      value: values[id]
    }
  }

  render() {
    const {
      classes,
      list,
      intl,
      setFieldValue,
      values,
      handleSubmit,
      isLoading
    } = this.props

    return (
      <Grid container spacing={16} alignItems="center" direction="column">
        {isLoading ? (
          <CircularProgress
            id="spinner"
            size={70}
            className={classes.spinner}
          />
        ) : (
          <Grid item xs={12} md={6} lg={4} className={classes.root}>
            <MaskedInput
              {...this.getProps('phone')}
              mask="+7 (999) 999-99-99"
              className={classes.input}
            />

            <MaskedInput
              {...this.getProps('sum')}
              mask="9999"
              className={classes.input}
              startAdornment={
                <InputAdornment position="start">&#x20bd;</InputAdornment>
              }
            />

            <MaskedInput
              {...this.getProps('card')}
              mask="9999 9999 9999 9999"
              className={classes.input}
            />

            <MaskedInput
              {...this.getProps('cvv')}
              mask="999"
              className={classes.input}
            />

            <TextField
              id="operator"
              select
              label={intl.formatMessage({ id: 'Form.operator' })}
              className={classes.input}
              value={values.operator}
              onChange={e => setFieldValue('operator', e.target.value)}
              margin="normal"
            >
              {list.map(option => (
                <MenuItem key={option.value} value={option.value}>
                  {intl.formatMessage({ id: `Operator.${option.value}` })}
                </MenuItem>
              ))}
            </TextField>

            <Button
              variant="contained"
              color="primary"
              className={classes.button}
              onClick={handleSubmit}
            >
              {intl.formatMessage({ id: 'Actions.pay' })}
              <Payment className={classes.icon} />
            </Button>
          </Grid>
        )}
      </Grid>
    )
  }
}

export default Form
