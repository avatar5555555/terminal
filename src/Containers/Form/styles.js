// @flow

import type { Theme } from 'src/types'

const styles = (theme: Theme) => ({
  root: {
    width: '100%',
    display: 'flex',
    justifyContent: 'space-between',
    flexWrap: 'wrap',
    margin: theme.spacing.unit,
    [theme.breakpoints.down('sm')]: {
      flexDirection: 'column',
      alignItems: 'flex-center'
    }
  },
  input: {
    margin: theme.spacing.unit,
    width: 200,
    [theme.breakpoints.down('sm')]: {
      width: '100%'
    }
  },
  button: {
    alignSelf: 'flex-end',
    margin: theme.spacing.unit,
    [theme.breakpoints.down('sm')]: {
      width: '94%'
    }
  },
  icon: {
    marginLeft: theme.spacing.unit
  },
  spinner: {
    marginTop: theme.spacing.unit * 10
  }
})

export default styles
