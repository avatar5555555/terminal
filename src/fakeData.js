const fakeData = [
  { name: 'mts', src: '/Images/mts.png' },
  { name: 'beeline', src: '/Images/beeline.png' },
  { name: 'megafon', src: '/Images/megafon.png' },
  { name: 'other', src: '/Images/other.png' }
]

export default fakeData
