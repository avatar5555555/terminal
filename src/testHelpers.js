// created consider react-intl wiki https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl
import React from 'react'
import { IntlProvider, intlShape } from 'react-intl'
import { mount, shallow } from 'enzyme'

import messages from 'src/i18n'
import config from 'src/config'

const { locale } = config
// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider(
  { locale, messages: messages[locale] },
  {}
)
const { intl } = intlProvider.getChildContext()

/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
const nodeWithIntlProp = node => React.cloneElement(node, { intl })

export const shallowWithIntl = (node, { context, ...additionalOptions } = {}) =>
  shallow(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    ...additionalOptions
  })

export const mountWithIntl = (
  node,
  { context, childContextTypes, ...additionalOptions } = {}
) =>
  mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    childContextTypes: Object.assign(
      {},
      { intl: intlShape },
      childContextTypes
    ),
    ...additionalOptions
  })
