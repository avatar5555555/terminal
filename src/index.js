import React from 'react'
import ReactDOM from 'react-dom'
import { IntlProvider } from 'react-intl'
import { BrowserRouter } from 'react-router-dom'
import { MuiThemeProvider } from '@material-ui/core/styles'
import { Provider } from 'react-redux'

import './index.css'

import App from './App'
import registerServiceWorker from './registerServiceWorker'

import messages from 'src/i18n'
import config from 'src/config'
import theme from 'src/services/theme'
import store from 'src/store'

const { locale } = config

const renderApp = () => (
  <IntlProvider
    locale={locale}
    defaultLocale={locale}
    messages={messages[locale]}
  >
    <BrowserRouter>
      <MuiThemeProvider theme={theme}>
        <Provider store={store}>
          <App />
        </Provider>
      </MuiThemeProvider>
    </BrowserRouter>
  </IntlProvider>
)

ReactDOM.render(renderApp(), document.getElementById('root'))
registerServiceWorker()
