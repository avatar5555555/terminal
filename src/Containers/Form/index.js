// @flow

import * as yup from 'yup'
import { withFormik } from 'formik'
import { compose } from 'redux'
import { connect } from 'react-redux'
import { injectIntl } from 'react-intl'
import { withStyles } from '@material-ui/core/styles'

import Form from './Form'
import styles from './styles'

import { isPhone } from 'src/services/validation'
import { getOperatorsList, getIsLoading } from 'src/store/selectors'
import { operatorsListReadRequest, paymentRequest } from 'src/store/actions'
import type { State, FormikProps, FormikBag, FormValues } from 'src/types'

export const mapPropsToValues = ({ location }: FormikProps) => {
  const { state } = location
  return {
    sum: '',
    phone: '',
    operator: state ? state.name : '',
    card: '',
    cvv: ''
  }
}

const validationSchema = yup.object().shape({
  phone: yup
    .string()
    .ensure()
    .test('check if phone valid', 'invalidPhone', isPhone)
    .required('required'),
  sum: yup
    .number()
    .min(1, 'minSum')
    .max(1000, 'maxSum')
    .required('required'),
  card: yup
    .number()
    .required('required')
    .min(12, 'invalidCard'),
  cvv: yup
    .string()
    .min(3, 'invalidCvv')
    .required('required')
})

export const handleSubmit = (values: FormValues, formikBag: FormikBag) => {
  const { pay } = formikBag.props

  pay(values)
}

export const mapState = (state: State) => {
  const operators = getOperatorsList(state)
  const selectOptions = operators.map(item => ({
    label: item.name,
    value: item.name
  }))

  return {
    list: selectOptions,
    isLoading: getIsLoading(state)
  }
}

export const bindActions = (dispatch: Function) => ({
  fetchList: () => dispatch(operatorsListReadRequest()),
  pay: () => dispatch(paymentRequest())
})

export default compose(
  injectIntl,
  connect(
    mapState,
    bindActions
  ),
  withFormik({
    mapPropsToValues,
    handleSubmit,
    validationSchema,
    validateOnBlur: false
  }),
  withStyles(styles)
)(Form)
