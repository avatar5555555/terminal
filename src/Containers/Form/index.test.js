import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import Form from './Form'
import styles from './styles'

import { mapPropsToValues, bindActions, mapState, handleSubmit } from '.'

import { mountWithIntl } from 'src/testHelpers'
import fakeData from 'src/fakeData'

const props = {
  errors: { phone: 'invalidPhone', sum: 'required' },
  touched: { phone: false, sum: true },
  values: { phone: 'string', sum: 'string', operator: 'string' },
  handleChange: jest.fn(),
  handleBlur: jest.fn(),
  fetchList: jest.fn(),
  handleSubmit: jest.fn(),
  setFieldValue: jest.fn(),
  intl: { formatMessage: jest.fn() },
  list: [{ label: 'mts', value: 'mts' }],
  classes: {
    root: 'string',
    input: 'string',
    button: 'string',
    icon: 'string'
  }
}

const mockStore = configureStore()
const initialState = {
  list: [],
  isLoading: false
}

describe('Form with state', () => {
  const wrap = (props = {}, store = {}) =>
    shallow(<Form store={store} {...props} />)

  it('renders', () => {
    const store = mockStore(initialState)
    expect(wrap(props, store)).toMatchSnapshot()
  })

  it('calls fetchList on componentDidMount', () => {
    const store = mockStore(initialState)
    const wrapper = mountWithIntl(<Form store={store} {...props} />)

    expect(wrapper).toBeDefined()
    expect(props.fetchList).toBeCalled()
  })
})

describe('Form', () => {
  const wrap = (props = {}) => shallow(<Form {...props} />)

  it('renders spinner if data is loading', () => {
    const wrapper = wrap({ ...props, isLoading: true })

    expect(wrapper.dive().find('#spinner')).toBeDefined()
  })

  it('calls setFieldValue onSelect', () => {
    const wrapper = wrap(props)

    wrapper
      .dive()
      .find('#operator')
      .simulate('change', { target: { value: 'hello' } })

    expect(props.setFieldValue.mock.calls.length).toBe(1)
  })
})

describe('Form functions', () => {
  it('styles returns object', () => {
    const obj = styles({
      spacing: { unit: 5 },
      breakpoints: { down: jest.fn() }
    })

    expect(obj.root.margin).toBe(5)
  })

  it('mapPropsToValues return obj with fulfilled operator', () => {
    const name = 'mts'
    const obj = mapPropsToValues({ location: { state: { name } } })

    expect(obj.operator).toBe(name)
  })

  it('mapPropsToValues return obj with empty operator', () => {
    const obj = mapPropsToValues({ location: {} })

    expect(obj.operator).toBe('')
  })

  it('handleSubmit calls pay with values and pass it into', () => {
    const mockNf = jest.fn()
    const operator = 'mts'
    handleSubmit({ operator }, { props: { pay: mockNf } })

    expect(mockNf.mock.calls.length).toBe(1)
    expect(mockNf.mock.calls[0][0]).toEqual({ operator })
  })

  it('mapState returns prepared list of operators', () => {
    const mappedProps = mapState({ ...initialState, list: fakeData })

    expect(mappedProps.list[0]).toEqual({ label: 'mts', value: 'mts' })
  })

  it('bindActions takes dispatch and returns object with fn', () => {
    const mockDispatch = jest.fn()

    const actions = bindActions(mockDispatch)

    actions.fetchList()
    actions.pay()

    expect(mockDispatch.mock.calls.length).toBe(2)
  })
})
