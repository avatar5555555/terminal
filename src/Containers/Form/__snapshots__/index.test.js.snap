// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form with state renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    classes={
        Object {
            "button": "string",
            "icon": "string",
            "input": "string",
            "root": "string",
          }
    }
    errors={
        Object {
            "phone": "invalidPhone",
            "sum": "required",
          }
    }
    fetchList={[Function]}
    handleBlur={[Function]}
    handleChange={[Function]}
    handleSubmit={[Function]}
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    list={
        Array [
            Object {
              "label": "mts",
              "value": "mts",
            },
          ]
    }
    setFieldValue={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    touched={
        Object {
            "phone": false,
            "sum": true,
          }
    }
    values={
        Object {
            "operator": "string",
            "phone": "string",
            "sum": "string",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "center",
      "children": <WithStyles(Grid)
        className="string"
        item={true}
        lg={4}
        md={6}
        xs={12}
>
        <MaskedInput
                className="string"
                error="invalidPhone"
                id="phone"
                mask="+7 (999) 999-99-99"
                onBlur={[Function]}
                onChange={[Function]}
                valid={true}
                value="string"
        />
        <MaskedInput
                className="string"
                error="required"
                id="sum"
                mask="9999"
                onBlur={[Function]}
                onChange={[Function]}
                startAdornment={
                        <WithStyles(InputAdornment)
                                position="start"
                        >
                                ₽
                        </WithStyles(InputAdornment)>
                }
                valid={false}
                value="string"
        />
        <MaskedInput
                className="string"
                error={undefined}
                id="card"
                mask="9999 9999 9999 9999"
                onBlur={[Function]}
                onChange={[Function]}
                valid={true}
                value={undefined}
        />
        <MaskedInput
                className="string"
                error={undefined}
                id="cvv"
                mask="999"
                onBlur={[Function]}
                onChange={[Function]}
                valid={true}
                value={undefined}
        />
        <TextField
                className="string"
                id="operator"
                label={undefined}
                margin="normal"
                onChange={[Function]}
                required={false}
                select={true}
                value="string"
        >
                <WithStyles(MenuItem)
                        value="mts"
                />
        </TextField>
        <WithStyles(Button)
                className="string"
                color="primary"
                onClick={[Function]}
                variant="contained"
        >
                <pure(Payment)
                        className="string"
                />
        </WithStyles(Button)>
</WithStyles(Grid)>,
      "container": true,
      "direction": "column",
      "spacing": 16,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MaskedInput
            className="string"
            error="invalidPhone"
            id="phone"
            mask="+7 (999) 999-99-99"
            onBlur={[Function]}
            onChange={[Function]}
            valid={true}
            value="string"
/>,
          <MaskedInput
            className="string"
            error="required"
            id="sum"
            mask="9999"
            onBlur={[Function]}
            onChange={[Function]}
            startAdornment={
                        <WithStyles(InputAdornment)
                                    position="start"
                        >
                                    ₽
                        </WithStyles(InputAdornment)>
            }
            valid={false}
            value="string"
/>,
          <MaskedInput
            className="string"
            error={undefined}
            id="card"
            mask="9999 9999 9999 9999"
            onBlur={[Function]}
            onChange={[Function]}
            valid={true}
            value={undefined}
/>,
          <MaskedInput
            className="string"
            error={undefined}
            id="cvv"
            mask="999"
            onBlur={[Function]}
            onChange={[Function]}
            valid={true}
            value={undefined}
/>,
          <TextField
            className="string"
            id="operator"
            label={undefined}
            margin="normal"
            onChange={[Function]}
            required={false}
            select={true}
            value="string"
>
            <WithStyles(MenuItem)
                        value="mts"
            />
</TextField>,
          <WithStyles(Button)
            className="string"
            color="primary"
            onClick={[Function]}
            variant="contained"
>
            <pure(Payment)
                        className="string"
            />
</WithStyles(Button)>,
        ],
        "className": "string",
        "item": true,
        "lg": 4,
        "md": 6,
        "xs": 12,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "string",
            "error": "invalidPhone",
            "id": "phone",
            "mask": "+7 (999) 999-99-99",
            "onBlur": [Function],
            "onChange": [Function],
            "valid": true,
            "value": "string",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "string",
            "error": "required",
            "id": "sum",
            "mask": "9999",
            "onBlur": [Function],
            "onChange": [Function],
            "startAdornment": <WithStyles(InputAdornment)
              position="start"
>
              ₽
</WithStyles(InputAdornment)>,
            "valid": false,
            "value": "string",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "string",
            "error": undefined,
            "id": "card",
            "mask": "9999 9999 9999 9999",
            "onBlur": [Function],
            "onChange": [Function],
            "valid": true,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "string",
            "error": undefined,
            "id": "cvv",
            "mask": "999",
            "onBlur": [Function],
            "onChange": [Function],
            "valid": true,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <WithStyles(MenuItem)
                value="mts"
/>,
            ],
            "className": "string",
            "id": "operator",
            "label": undefined,
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": true,
            "value": "string",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "mts",
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "value": "mts",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              undefined,
              <pure(Payment)
                className="string"
/>,
            ],
            "className": "string",
            "color": "primary",
            "onClick": [Function],
            "variant": "contained",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "string",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "center",
        "children": <WithStyles(Grid)
          className="string"
          item={true}
          lg={4}
          md={6}
          xs={12}
>
          <MaskedInput
                    className="string"
                    error="invalidPhone"
                    id="phone"
                    mask="+7 (999) 999-99-99"
                    onBlur={[Function]}
                    onChange={[Function]}
                    valid={true}
                    value="string"
          />
          <MaskedInput
                    className="string"
                    error="required"
                    id="sum"
                    mask="9999"
                    onBlur={[Function]}
                    onChange={[Function]}
                    startAdornment={
                              <WithStyles(InputAdornment)
                                        position="start"
                              >
                                        ₽
                              </WithStyles(InputAdornment)>
                    }
                    valid={false}
                    value="string"
          />
          <MaskedInput
                    className="string"
                    error={undefined}
                    id="card"
                    mask="9999 9999 9999 9999"
                    onBlur={[Function]}
                    onChange={[Function]}
                    valid={true}
                    value={undefined}
          />
          <MaskedInput
                    className="string"
                    error={undefined}
                    id="cvv"
                    mask="999"
                    onBlur={[Function]}
                    onChange={[Function]}
                    valid={true}
                    value={undefined}
          />
          <TextField
                    className="string"
                    id="operator"
                    label={undefined}
                    margin="normal"
                    onChange={[Function]}
                    required={false}
                    select={true}
                    value="string"
          >
                    <WithStyles(MenuItem)
                              value="mts"
                    />
          </TextField>
          <WithStyles(Button)
                    className="string"
                    color="primary"
                    onClick={[Function]}
                    variant="contained"
          >
                    <pure(Payment)
                              className="string"
                    />
          </WithStyles(Button)>
</WithStyles(Grid)>,
        "container": true,
        "direction": "column",
        "spacing": 16,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MaskedInput
              className="string"
              error="invalidPhone"
              id="phone"
              mask="+7 (999) 999-99-99"
              onBlur={[Function]}
              onChange={[Function]}
              valid={true}
              value="string"
/>,
            <MaskedInput
              className="string"
              error="required"
              id="sum"
              mask="9999"
              onBlur={[Function]}
              onChange={[Function]}
              startAdornment={
                            <WithStyles(InputAdornment)
                                          position="start"
                            >
                                          ₽
                            </WithStyles(InputAdornment)>
              }
              valid={false}
              value="string"
/>,
            <MaskedInput
              className="string"
              error={undefined}
              id="card"
              mask="9999 9999 9999 9999"
              onBlur={[Function]}
              onChange={[Function]}
              valid={true}
              value={undefined}
/>,
            <MaskedInput
              className="string"
              error={undefined}
              id="cvv"
              mask="999"
              onBlur={[Function]}
              onChange={[Function]}
              valid={true}
              value={undefined}
/>,
            <TextField
              className="string"
              id="operator"
              label={undefined}
              margin="normal"
              onChange={[Function]}
              required={false}
              select={true}
              value="string"
>
              <WithStyles(MenuItem)
                            value="mts"
              />
</TextField>,
            <WithStyles(Button)
              className="string"
              color="primary"
              onClick={[Function]}
              variant="contained"
>
              <pure(Payment)
                            className="string"
              />
</WithStyles(Button)>,
          ],
          "className": "string",
          "item": true,
          "lg": 4,
          "md": 6,
          "xs": 12,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "string",
              "error": "invalidPhone",
              "id": "phone",
              "mask": "+7 (999) 999-99-99",
              "onBlur": [Function],
              "onChange": [Function],
              "valid": true,
              "value": "string",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "string",
              "error": "required",
              "id": "sum",
              "mask": "9999",
              "onBlur": [Function],
              "onChange": [Function],
              "startAdornment": <WithStyles(InputAdornment)
                position="start"
>
                ₽
</WithStyles(InputAdornment)>,
              "valid": false,
              "value": "string",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "string",
              "error": undefined,
              "id": "card",
              "mask": "9999 9999 9999 9999",
              "onBlur": [Function],
              "onChange": [Function],
              "valid": true,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "string",
              "error": undefined,
              "id": "cvv",
              "mask": "999",
              "onBlur": [Function],
              "onChange": [Function],
              "valid": true,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <WithStyles(MenuItem)
                  value="mts"
/>,
              ],
              "className": "string",
              "id": "operator",
              "label": undefined,
              "margin": "normal",
              "onChange": [Function],
              "required": false,
              "select": true,
              "value": "string",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "mts",
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                  "value": "mts",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                undefined,
                <pure(Payment)
                  className="string"
/>,
              ],
              "className": "string",
              "color": "primary",
              "onClick": [Function],
              "variant": "contained",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "string",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
